coaster/views.py

```diff
@@ -575,26 +575,58 @@ def decorated_function(*args, **kwargs):
         return decorated_function
     return inner
 
-def cors(check_origin,
+def cors(allowed_origin,
     methods=['HEAD', 'OPTIONS', 'GET', 'POST', 'PUT', 'PATCH', 'DELETE'],
     headers=['Accept', 'Accept-Language', 'Content-Language', 'Content-Type', 'X-Requested-With'],
     max_age=None):
     """
     Adds CORS headers to the decorated view function.
 
-    :param check_origin: A function that receives the origin as a parameter and
-    is expected to return a boolean value to assert if the given origin has access to the
-    requested resource
+    :param allowed_origin: Either of (a) a callable that receives the origin as a parameter. The callable
+    is expected to check if the given origin has access to the
+    requested resource and return a boolean value (b) A list of origins
+    (c) '*', indicating that this resource is accessible by any origin
     :param methods: A list of HTTP methods that are allowed for this origin
     :param headers: A list of HTTP headers that are allowed for this origin
     :param max_age: Maximum number of seconds the result for the pre-flight request can be cached
+
+    Example use::
+        from flask import Flask, Response
+        from coaster.views import cors
+
+        app = Flask(__name__)
+
+        @app.route('/any')
+        @cors('*')
+        def any_origin():
+            return Response()
+
+        @app.route('/static', methods=['GET', 'POST'])
+        @cors(['https://hasgeek.com'], methods=['GET'], headers=['Content-Type', 'X-Requested-With'],
```

Comment:
Since the origin provided by the browser includes the scheme, it makes it easier to compare.